src/js_types.hpp

```diff
@@ -221,6 +223,24 @@ struct Function {
     static ObjectType construct(ContextType ctx, const FunctionType &function, const std::vector<ValueType> &arguments) {
         return construct(ctx, function, arguments.size(), arguments.data());
     }
+
+    /**
+     * These wrap a js callback into a C++ callback that takes an Optional<AppError> and possibly a result.
+     * The versions that accept a result use the converter argument to convert the C++ result into a js result.
+     * The returned callbacks have the following signatures:
+     *
+     * wrap_void_callback         - void(const util::Optional<app::AppError>& error)
+     * wrap_callback_error_first  - void(const util::Optional<app::AppError>& error, auto&& result)
+     * wrap_callback_result_first - void(auto&& result, const util::Optional<app::AppError>& error)
```

Comment:
Why don't we fix it in object store?

Reply:
I didn't want to break compile for everyone in order to make the callback argument order consistent. At least not without discussing it with the other sdk teams. I'll ping the slack channel and see what everyone thinks. Do you mind going with this for now, and if we fix obj store, we can eliminate one of the `wrap_callback_foo_first` functions in a later commit?